src/YesSql.Core/Provider/IDbProviderOptions.cs

```diff
@@ -0,0 +1,11 @@
+﻿using YesSql.Core.Services;
+
+namespace YesSql.Core.Provider
+{
+    public interface IDbProviderOptions
+    {
+        string ProviderName { get; set; }
```

Comment:
> if that's the case I think Orchard CMS is already have a way to list those providers from DI or some where else, which is out of the scope here

Yes, so you can remove the `ProviderName` option

> > Also where do you specify that a provider accepts connection strings or not?

> this is done in the extension methods UseSqlServer("connectionString")

You quote me, but you missunderstand this quote. Yes the extension method accepts a connection string. But in Orchard CMS if a provider doesn't need a connection string to be provided by the user we don't display the text box for it in the UI. I am talking "UI metadata" not actual method call.